src/controller/CHIPDeviceController.h

```diff
@@ -460,13 +461,67 @@ class DLL_EXPORT DeviceCommissioner : public DeviceController, public SessionEst
 
     static void OnSessionEstablishmentTimeoutCallback(System::Layer * aLayer, void * aAppState, System::Error aError);
 
-    CHIP_ERROR SendOperationalCertificateSigningRequestCommand(NodeId remoteDeviceId);
-    CHIP_ERROR OnOperationalCertificateSigningRequest(NodeId node, const ByteSpan & csr);
-    CHIP_ERROR SendOperationalCertificate(NodeId remoteDeviceId, const ByteSpan & opCertBuf, const ByteSpan & icaCertBuf);
-    CHIP_ERROR SendTrustedRootCertificate(NodeId remoteDeviceId, const ByteSpan & certBuf);
+    /* This function sends an OpCSR request to the device */
+    CHIP_ERROR SendOperationalCertificateSigningRequestCommand(Device * device);
+    /* This function sends the operational credentials to the device */
+    CHIP_ERROR SendOperationalCertificate(Device * device, const ByteSpan & opCertBuf, const ByteSpan & icaCertBuf);
+    /* This function sends the trusted root certificate to the device */
+    CHIP_ERROR SendTrustedRootCertificate(Device * device);
+
+    /* This function is called when the device completes the operational credential provisioning process */
+    CHIP_ERROR OnOperationalCredentialsProvisioningCompletion(Device * device);
+
+    /* This function is called when the previously sent CSR request results in failure */
+    static void OnCSRFailureResponse(void * context, uint8_t status);
+
+    /**
+     * @brief
+     *   This function is called by the IM layer when the commissioner receives the CSR from the device.
+     *
+     * @param[in] context         The context provided while registering the callback.
+     * @param[in] CSR             The Certificate Signing Request.
+     * @param[in] CSRNonce        The Nonce sent by us when we requested the CSR.
+     * @param[in] VendorReserved1 vendor-specific information that may aid in device commissioning.
+     * @param[in] VendorReserved2 vendor-specific information that may aid in device commissioning.
+     * @param[in] VendorReserved3 vendor-specific information that may aid in device commissioning.
+     * @param[in] Signature       Cryptographic signature generated for the fields in the response message.
```

Comment:
It might be helpful to point to which parts of the spec these things correspond to.